payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -89,13 +91,40 @@ internal class DefaultCardNumberController constructor(
 
     private val isEligibleForCardBrandChoice = cardBrandChoiceConfig is CardBrandChoiceConfig.Eligible
     private val brandChoices = MutableStateFlow<List<CardBrand>>(listOf())
-    private val chosenBrand = MutableStateFlow(
+
+    private val preferredBrands = when (cardBrandChoiceConfig) {
+        is CardBrandChoiceConfig.Eligible -> cardBrandChoiceConfig.preferredBrands
+        is CardBrandChoiceConfig.Ineligible -> listOf()
+    }
+
+    /*
+     * This flow is keeping track of whatever brand the user had selected from the dropdown menu
+     * regardless of whether their card number has changed.
+     *
+     * This will allow us re-reference the previously selected choice if the user changes the card
+     * number.
+     */
+    private val uncheckedChosenBrand = MutableStateFlow(
```

Comment:
The naming here throws me off a bit. Should this be `userSelectedBrand`? And if so, why does it use `initialBrand`, which doesn’t come from the user?

Reply:
`initialBrand` can either be the user choice that they selected initially (ie. adding a card in flow controller then going back to edit it) or also the preferred merchant choice. I think the comment itself should be changed to reflect that it isn't just user choice. I could make the naming `unsanitizedBrand` instead?